### PR TITLE
Create a common telemetry plugin

### PR DESCRIPTION
**Pull request by soup-of-the-day**
### What this PR does / why we need it
TKG currently has a unique telemetry mechanism while other products (such as TAP) are adopting the telemetry SDK. This will lead to divergent experiences with regards to telemetry on TKG. So, a telemetry plugin was proposed and accepted through a TKG design review.

This PR creates that plugin for managing telemetry settings on any cluster with tanzu products. This is intended for use in configuring products that use the telemetry SDK. Wanted to open this PR early to get high level feedback on the code I have so far. It functionally works, but I'm not as familiar with this code base and wanted to get early input.

In parallel to this work, I'll also work on changes to the TKG telemetry image to use the SDK when sending data. The outcome should be a telemetry image that respects the SDK's telemetry settings (including whether or not the cluster has been opted into telemetry). This will mean that the management-cluster plugin will no longer manage CEIP opt-in/out, and should be deprecated.

### Describe testing done for PR

- Currently  have extensive unit tests for new code. 
- Manual testing has been done on a local cluster for functionality.

### Release note

```release-note
Added a new telemetry subcommand for managing cluster-wide telemetry settings.
```

### PR Checklist

- [ ] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

#### Special notes for your reviewer

In addition to general feedback, it'd be nice to get some input on:
- tanzu-framework integration testing in general
- how tanzu CLI should handle targeting clusters, just using a KUBECONFIG env var right now

### CODE REVIEW

**Comment by stmcginnis** [comment]
Please edit proposal link to something publicly accessible.

File: cmd/cli/plugin/telemetry/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    # opt into ceip
+    tanzu telemetry update --ceip-opt-in
+	# opt out of ceip
+    tanzu telemetry update --ceip-opt-out
+	# update shared configuration settings
+    tanzu telemetry update --env-is-prod "true" --entitlement-account-number "1234" --csp-org-id "XXXX"
+
+
+Flags:
+      --ceip-opt-in                         opt into VMware's ceip program
```
Comment (AlainRoy) [inline cmd/cli/plugin/telemetry/README.md:57]: CEIP is an acronym -- I recommend capitalizing.

**Comment by soup-of-the-day** [comment]
New changes since PR was opened a while ago:
1. Found out the telemetry plugin could be made context aware. Ideally the plugin would only exist for operators targeting TKG clusters v1.6.0 or later
2. the management-cluster plugin is NOT context-aware, so needed to add logic to have it cede control to the telemetry plugin where applicable. This means having the CEIP command check cluster version before doing anything, and printing out a deprecation notice on management clusters v1.6.0 or later
3. Since the TKG image is being updated to respect settings configured by the telemetry plugin, we need to enforce that the image is re-installed when users upgrade their clusters.
4. Rebased the PR since its been a while since changes were made.

File: packages/management/core-management-plugins/bundle/config/telemetry.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#@ load("@ytt:data", "data")
+#@ load("@ytt:overlay", "overlay")
+
+---
+apiVersion: cli.tanzu.vmware.com/v1alpha1
```
Comment (vuil) [inline packages/management/core-management-plugins/bundle/config/telemetry.yaml:5]: @anujc25 can you PTAL to see if this is impacted in any way by the package-based-lcm changes?

File: common.mk [hunk 3]
```diff
@@ -41,7 +41,7 @@ LD_FLAGS += -X 'github.com/vmware-tanzu/tanzu-framework/pkg/v1/buildinfo.Version
 
 # Add supported OS-ARCHITECTURE combinations here
 ENVS ?= linux-amd64 windows-amd64 darwin-amd64
-STANDALONE_PLUGINS ?= login management-cluster package pinniped-auth secret
+STANDALONE_PLUGINS ?= login management-cluster package pinniped-auth secret telemetry
```
Comment (anujc25) [inline common.mk:44]: Considering this is a context-scoped plugin, it should be added to the next array `CONTEXTAWARE_PLUGINS`.

**Comment by anujc25** [comment]
There are few linter errors that are failing `Lint` pipeline.

**Review by anujc25 — APPROVED** [review]
LGTM

**Pull request merged**